odoo/tools/translate.py

```diff
@@ -888,6 +889,45 @@ def handle_text(text, lineno):
     _extract_translatable_qweb_terms(tree.getroot(), handle_text)
     return result
 
+
+def extract_spreadsheet_terms(fileobj, keywords, comment_tags, options):
+    """Babel message extractor for spreadsheet data files.
+
+    :param fileobj: the file-like object the messages should be extracted from
+    :param keywords: a list of keywords (i.e. function names) that should
+                     be recognized as translation functions
+    :param comment_tags: a list of translator tags to search for and
+                         include in the results
+    :param options: a dictionary of additional options (optional)
+    :return: an iterator over ``(lineno, funcname, message, comments)``
+             tuples
+    :rtype: Iterable
+    """
+    terms = []
+    data = json.load(fileobj)
+    for sheet in data.get('sheets', []):
+        for cell in sheet['cells'].values():
+            content = cell.get('content', '')
+            terms += re.findall(r'_t\("(.+?)"\)', content, re.IGNORECASE)
```

Comment:
this can be dangerous: 
![Screenshot_20220923_110925](https://user-images.githubusercontent.com/7260798/191928375-b860e7db-d742-4f43-8800-ff8600bf4392.png)